### PR TITLE
fix: don't assume bash when displaying interactive CLI prompts

### LINKED ISSUE #21838: Escape characters break interactive prompts in `influxd upgrade` in PowerShell
__Steps to reproduce:__
Originally reported in #21534.

1. Run `influxd upgrade` from PowerShell

__Expected behavior:__
The command works the same as in bash.

__Actual behavior:__
1. Bash escape characters are printed to the screen
2. The password-prompt always fails to complete because the escape characters & newline we've hard-coded get registered as the re-entry of the password

The new code in `influx-cli` seems to have fixed this problem for its own interactive prompts. Hopefully reusing its utilities will help us here.

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #21838

~Requires influxdata/influx-cli#192 to merge first so we can re-update `go.mod` to point at `main`~

### CODE REVIEW

**Comment by danxmoran** [comment]
Ugh, now needs influxdata/influx-cli#195

File: cmd/influxd/upgrade/setup.go [hunk 1]
```diff
@@ -25,81 +22,35 @@ func setupAdmin(ctx context.Context, v2 *influxDBv2, req *influxdb.OnboardingReq
...
-		req.Password = internal.GetPassword(ui, false)
+func onboardingRequest(cli clients.CLI, options *options) (*influxdb.OnboardingRequest, error) {
+	setupClient := setup.Client{CLI: cli}
+	cliReq, err := setupClient.OnboardingRequest(&setup.Params{
+		Username:  options.target.userName,
+		Password:  options.target.password,
+		AuthToken: options.target.token,
+		Org:       options.target.orgName,
+		Bucket:    options.target.bucket,
+		Retention: options.target.retention,
```
Comment (lesam) [inline cmd/influxd/upgrade/setup.go:33]: It's a bit confusing that the prompt appears to be for a different kind of string than the options value now? (nanos vs hours?)

File: cmd/influxd/upgrade/setup.go [hunk 2]
```diff
@@ -25,81 +22,35 @@ func setupAdmin(ctx context.Context, v2 *influxDBv2, req *influxdb.OnboardingReq
...
-	if req.Org == "" {
-		req.Org = internal.GetInput(ui, "Please type your primary organization name", "")
+	req := influxdb.OnboardingRequest{
+		User:   cliReq.Username,
+		Org:    cliReq.Org,
+		Bucket: cliReq.Bucket,
 	}
-	if req.Bucket == "" {
-		req.Bucket = internal.GetInput(ui, "Please type your primary bucket name", "")
+	if cliReq.Password != nil {
```
Comment (lesam) [inline cmd/influxd/upgrade/setup.go:44]: So for `influxdb.OnboardingRequest` we don't distinguish between nil and default-value, but for `api.OnboardingRequest` we do?

Comment (danxmoran) [reply cmd/influxd/upgrade/setup.go:44]: Yes, a consequence of codegen vs. hand-rolled. The codegen'd models use pointer-fields for every optional field.

File: cmd/influxd/upgrade/setup.go (hunk 1 above)
Comment (danxmoran) [reply cmd/influxd/upgrade/setup.go:33]: The help-text isn't very helpful, but I think the behavior is the same before and after this change. Whatever gets passed to `--retention` gets parsed as a `time.Duration`
* In the old impl: https://github.com/influxdata/influxdb/pull/21839/files#diff-ec11f92a3df5beb5cc1657892bfc169a49bed091ce94ebd1fae3d989a8c6eceeL43
* In the new impl: https://github.com/influxdata/influx-cli/blob/main/clients/setup/setup.go#L141

I can update the description in the help-text to show examples of durations (i.e. `72h`)

File: cmd/influxd/upgrade/setup.go (hunk 2 above)
Comment (danxmoran) [reply cmd/influxd/upgrade/setup.go:44]: Talked on Slack: going to try deleting `influxdb.OnboardingRequest` and migrating existing code to use the codegen'd model imported from the CLI

Comment (danxmoran) [reply cmd/influxd/upgrade/setup.go:44]: After trying, would prefer to leave it as an adapter layer here. Both `influxdb.OnboardingRequest` and `api.OnboardingRequest` embed references to other models from their respective packages; if you change the top-level type, you also have to change how you process all the fields, and it starts to sprawl into many files.

**Comment by danxmoran** [comment]
Reset and force-pushed to clean up the git history, not sure what happened with all the merges from master

**Review by lesam — APPROVED** [review]

**Pull request merged**